framework/forms/__init__.py

```diff
@@ -56,7 +56,8 @@ def __init__(self, message=None):
         self.message = message or 'HTML is not allowed in form field'
 
     def __call__(self, form, field):
-        if not field.data == strip_html(field.data):
+        field_data = field.data or ''  # we do not aim to check equality, we aim to verify absence of HTML
+        if field_data != strip_html(field.data):
```

Comment:
I've just launched this test on pre python upgrade version of osf, and can confirm that we are facing second case. ALthrough, this validator's solely purpose is to prevent submission of any html characters, and if field value is None -> therefore there aren't any html characters